src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -224,7 +224,7 @@ public void ensureCapacity(int minimumCapacity) {
      */
     private void ensureCapacityInternal(int minimumCapacity) {
         // overflow-conscious code
-        int oldCapacity = value.length >> coder;
+        int oldCapacity = capacity();
```

Comment:
This doesn't seem related to the intent of the patch, and might distort startup profiles a bit.

Reply:
Reverted